rotkehlchen/chain/manager.py

```diff
@@ -621,10 +621,28 @@ def query_balances(
             self.query_polkadot_balances(ignore_cache=ignore_cache)
         if should_query_avax:
             self.query_avax_balances(ignore_cache=ignore_cache)
+        if ignore_cache is True:
+            self.derive_new_addresses_from_xpubs(
+                should_derive_bch_xpubs=should_query_bch,
+                should_derive_btc_xpubs=should_query_btc,
+            )
 
         self.totals = self.balances.recalculate_totals()
```

Comment:
@yabirgb what if no new addresses are derived? This would mean that there is a chance that we don't recalculate totals unless I am getting something incorrectly

Reply:
When does happen that the totals need to be recalculated if no new address is detected?

Reply:
Balances of all derived addresses are always updated (since we query them anyway when deriving addresses)

https://github.com/rotki/rotki/blob/develop/rotkehlchen/chain/bitcoin/xpub.py#L259

Reply:
Okey guess it is fine then